### PR TITLE
Move LinkTo component to a separate `addon-links/react` endpoint

### PR DESCRIPTION
**Pull request by Hypnosphi**
Issue: #2096 

## What I did
Moved react/browser-specific stuff to a separate endpoint

## How to test
Run crna-kitchen-sink

### CODE REVIEW

**Comment by tmeasday** [comment]
LGTM!

**Review by tmeasday — APPROVED** [review]

**Comment by shilman** [comment]
Is this a breaking change? Can we keep the old behavior & deprecate it until 4.0?

**Comment by Hypnosphi** [comment]
The feature was introduced only in prerelease, so we're free to change it. And the bug I fix here blocks storybook for RN from working

File: examples/cra-kitchen-sink/src/stories/welcome.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import React from 'react';
 import { Welcome } from '@storybook/react/demo';
 import { storiesOf } from '@storybook/react';
-import { linkTo } from '@storybook/addon-links';
 
-storiesOf('Welcome', module).add('to Storybook', () => <Welcome showApp={linkTo('Button')} />);
+storiesOf('Welcome', module).add('to Storybook', () => <Welcome showKind="Button" />);
```
Comment (danielduan) [inline examples/cra-kitchen-sink/src/stories/welcome.js]: do we need to update these in the cli templates?

**Review by danielduan — CHANGES_REQUESTED** [review]

File: examples/cra-kitchen-sink/src/stories/welcome.js (hunk 1 above)
Comment (Hypnosphi) [reply examples/cra-kitchen-sink/src/stories/welcome.js]: This was already updated. Actually I just did some copy-pasting from commented code in `stories/index.js` to partially restore the `release/3.3` changes.

If you feel like it, you can sync the rest of the changes in another PR

Comment (ndelangen) [reply examples/cra-kitchen-sink/src/stories/welcome.js]: Why does this need to change to begin with?

**Review by danielduan — APPROVED** [review]

File: examples/cra-kitchen-sink/src/stories/welcome.js (hunk 1 above)
Comment (Hypnosphi) [reply examples/cra-kitchen-sink/src/stories/welcome.js]: It has changed in #1829 not now. It’basically adding a usage of a new feature

Comment (Hypnosphi) [reply examples/cra-kitchen-sink/src/stories/welcome.js]: It has changed in #1829 not now. It’basically adding a usage of a new feature

Comment (ndelangen) [reply examples/cra-kitchen-sink/src/stories/welcome.js]: Ok, well I'm OK with this change.

**Review by ndelangen — APPROVED** [review]

**Pull request merged**